File: 1/output.js
const fs = require("fs");

fs.readFile("input.txt", "utf8", (err, data) => {
  if (err) {
    console.error(err);
    return;
  }

  // Part one
  let max = 0;
  let localMax = 0;

  for (const calories of data.split("\n")) {
    if (calories === "") {
      max = Math.max(localMax, max);
      localMax = 0;
    } else {
      localMax += parseInt(calories);
    }
  }

  console.log(max);

  //Part two
  const elves = [];
  let elf = 0;
  for (const calories of data.split("\n")) {
    if (calories === "") {
      elves.push(elf);
      elf = 0;
    } else {
      elf += parseInt(calories);
    }
  }

  console.log(
    elves
      .sort((a, b) => b - a)
      .slice(0, 3)
      .reduce((prev, curr) => prev + curr)
  );
});

File: 6-Ruby/output.rb
file = File.open "input.txt"
sequence = file.read.split ''

marker = []
i = 0

# Part one

for c in sequence do
    i += 1
    marker.append c
    if marker.length == 4
        if marker.uniq.length == marker.length
            puts i
            break
        else
            marker.shift
        end
    end
end

# Part two

marker = []
i = 0

for c in sequence do
    i += 1
    marker.append c
    if marker.length == 14
        if marker.uniq.length == marker.length
            puts i
            break
        else
            marker.shift
        end
    end
end